src/components/Articles.jsx

```diff
@@ -1,34 +1,46 @@
-import {useParams} from 'react-router-dom';
+import {useParams,Link} from 'react-router-dom';
 import {useState, useEffect} from 'react';
 import axios from 'axios';
+import Pagination from './Pagination';
 
 function Articles() {
     const params = useParams();
     console.log(params);
     const [articles, setArticles] = useState([]);
+    const [isLoading , setIsLoading] = useState(false);
```

Comment:
default for loading states makes more sense to be true. It is loading until the first data call completes in the useEffect.